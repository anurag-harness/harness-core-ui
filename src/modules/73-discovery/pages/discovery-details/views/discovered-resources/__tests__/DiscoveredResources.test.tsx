/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { act, fireEvent, render, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import * as servicediscovery from 'services/servicediscovery'
import { TestWrapper } from '@common/utils/testUtils'
import DiscoveredResources from '../DiscoveredResources'
import { mockNamespaces, mockConnections } from '../mocks'

jest.mock('services/servicediscovery', () => ({
  useListDiscoveredServiceConnection: jest.fn().mockImplementation(() => {
    return { data: mockConnections, refetch: jest.fn(), error: null, loading: false }
  }),
  useListNamespace: jest.fn().mockImplementation(() => {
    return { data: mockNamespaces, refetch: jest.fn(), error: null, loading: false }
  })
}))

jest.mock('../graphView/DiscoveredResourcesGraph', () => ({
  ...jest.requireActual('../graphView/DiscoveredResourcesGraph'),
  __esModule: true,
  default: () => {
    return <div className={'discoveredResourcesGraph'}>Discovered Resources Graph</div>
  }
}))
jest.mock('../tableView/DiscoveredServices', () => ({
  ...jest.requireActual('../tableView/DiscoveredServices'),
  __esModule: true,
  default: () => {
    return <div className={'discoveredServices'}>Discovered Services</div>
  }
}))

describe('DiscoveredResources', () => {
  test('render component', async () => {
    const { container } = render(
      <TestWrapper>
        <DiscoveredResources />
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })

  test('subheader components should be present', async () => {
    const { container, getByTestId, getByPlaceholderText } = render(
      <TestWrapper>
        <DiscoveredResources />
      </TestWrapper>
    )

    const dropdown = getByTestId('namespace')
    act(() => {
      fireEvent.click(dropdown)
    })

    const listItem = container.getElementsByClassName('DropDown--menuItem')[0]
    act(() => {
      fireEvent.click(listItem)
    })

    const query = 'test abc'
    const searchInput = getByPlaceholderText('discovery.searchService') as HTMLInputElement
    expect(searchInput).not.toBe(null)
    if (!searchInput) {
      throw Error('no search input')
    }
    await userEvent.type(searchInput, query)
    await waitFor(() => expect(searchInput?.value).toBe(query))

    expect(container).toMatchSnapshot()
  })

  test('Toggle between grid and list view', async () => {
    const { container } = render(
      <TestWrapper>
        <DiscoveredResources />
      </TestWrapper>
    )
    const list = container.querySelectorAll("[icon='list']")[0]
    const grid = container.querySelectorAll("[icon='grid-view']")[0]
    await act(async () => {
      fireEvent.click(list!)
      expect(container).toMatchSnapshot()
      fireEvent.click(grid!)
      expect(container).toMatchSnapshot()
    })
  })

  test('when namespace data is undefined', async () => {
    jest.spyOn(servicediscovery, 'useListNamespace').mockImplementation((): any => {
      return {
        data: undefined,
        loading: false,
        error: {
          message: 'some error'
        }
      }
    })
    const { container } = render(
      <TestWrapper>
        <DiscoveredResources />
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })
})
