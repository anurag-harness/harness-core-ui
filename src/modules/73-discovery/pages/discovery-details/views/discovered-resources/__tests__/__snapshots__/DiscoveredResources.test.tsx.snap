// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiscoveredResources Toggle between grid and list view 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main"
  >
    <div
      class="Layout--horizontal StyledProps--main PageSubHeader--container StyledProps--flex"
      data-testid="page-subheader"
    >
      <div
        class="Layout--horizontal StyledProps--main StyledProps--flex"
        style="gap: 8px; width: 100%;"
      >
        <div
          class="bp3-popover-wrapper DropDown--main"
        >
          <div
            class="bp3-popover-target"
          >
            <div
              class=""
            >
              <div
                class="Layout--horizontal StyledProps--main DropDown--dropdownButton DropDown--withBorder StyledProps--flex"
                data-testid="namespace"
                style="width: 160px;"
              >
                <div
                  class="Layout--horizontal StyledProps--main DropDown--labelWrapper StyledProps--flex"
                >
                  <p
                    class="StyledProps--font Text--lineclamp StyledProps--main DropDown--label"
                    data-testid="dropdown-value"
                    style="--text-line-clamp: 1;"
                  >
                    common.namespace
                  </p>
                </div>
                <div
                  class="Layout--horizontal StyledProps--main DropDown--btnWrapper StyledProps--flex"
                >
                  <span
                    class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey400"
                    data-icon="main-chevron-down"
                  >
                    <svg
                      height="8"
                      viewBox="0 0 14 8"
                      width="8"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M13 0c-.3 0-.5.1-.7.3L7 5.6 1.7.3C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7l6 6c.2.2.4.3.7.3s.5-.1.7-.3l6-6c.2-.2.3-.4.3-.7 0-.6-.4-1-1-1z"
                        fill="#4d5969"
                      />
                    </svg>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <span
          style="flex-grow: 1;"
        />
        <div
          class="bp3-input-group ui-search-box ExpandingSearchInput--main   ExpandingSearchInput--alwaysExpanded ExpandingSearchInput--light"
          data-name=""
          style="width: 232px; max-width: 232px;"
        >
          <span
            class="bp3-icon StyledProps--main ExpandingSearchInput--icon StyledProps--color StyledProps--color-grey700"
            data-icon="thinner-search"
          >
            <svg
              fill="none"
              height="14"
              viewBox="0 0 16 16"
              width="14"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.125 13.25c1.617 0 3.32-.695 4.375-1.75L15 15l-3.5-3.5c1.021-1.005 1.75-2.894 1.75-4.375a6.125 6.125 0 10-6.125 6.125z"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </span>
          <input
            class="bp3-input  ExpandingSearchInput--filter"
            dir="auto"
            placeholder="discovery.searchService"
            style="padding-right: 38px;"
            type="search"
            value=""
          />
        </div>
        <div
          class="Layout--horizontal StyledProps--main StyledProps--flex"
        >
          <button
            class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main GridListToggle--gridButton StyledProps--intent-primary Button--with-current-color Button--without-shadow Button--withLeftIcon"
            data-testid="grid-view"
            data-tooltip-id="grid-view"
            type="button"
          >
            <span
              class="bp3-icon bp3-icon-grid-view StyledProps--main"
              icon="grid-view"
            >
              <svg
                data-icon="grid-view"
                height="16"
                viewBox="0 0 16 16"
                width="16"
              >
                <desc>
                  grid-view
                </desc>
                <path
                  d="M0 1v6h7V0H1C.45 0 0 .45 0 1zm0 14c0 .55.45 1 1 1h6V9H0v6zM15 0H9v7h7V1c0-.55-.45-1-1-1zM9 16h6c.55 0 1-.45 1-1V9H9v7z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </button>
          <button
            class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main GridListToggle--listUnselected GridListToggle--listButton Button--with-current-color Button--iconOnly Button--without-shadow Button--withLeftIcon"
            data-testid="list-view"
            data-tooltip-id="list-view"
            type="button"
          >
            <span
              class="bp3-icon bp3-icon-list StyledProps--main"
              icon="list"
            >
              <svg
                data-icon="list"
                height="16"
                viewBox="0 0 16 16"
                width="16"
              >
                <desc>
                  list
                </desc>
                <path
                  d="M1 3h14c.55 0 1-.45 1-1s-.45-1-1-1H1c-.55 0-1 .45-1 1s.45 1 1 1zm14 10H1c-.55 0-1 .45-1 1s.45 1 1 1h14c.55 0 1-.45 1-1s-.45-1-1-1zm0-4H1c-.55 0-1 .45-1 1s.45 1 1 1h14c.55 0 1-.45 1-1s-.45-1-1-1zm0-4H1c-.55 0-1 .45-1 1s.45 1 1 1h14c.55 0 1-.45 1-1s-.45-1-1-1z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="StyledProps--font StyledProps--main"
    >
      <div
        class="discoveredResourcesGraph"
      >
        Discovered Resources Graph
      </div>
    </div>
  </div>
</div>
`;

exports[`DiscoveredResources Toggle between grid and list view 2`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main"
  >
    <div
      class="Layout--horizontal StyledProps--main PageSubHeader--container StyledProps--flex"
      data-testid="page-subheader"
    >
      <div
        class="Layout--horizontal StyledProps--main StyledProps--flex"
        style="gap: 8px; width: 100%;"
      >
        <div
          class="bp3-popover-wrapper DropDown--main"
        >
          <div
            class="bp3-popover-target"
          >
            <div
              class=""
            >
              <div
                class="Layout--horizontal StyledProps--main DropDown--dropdownButton DropDown--withBorder StyledProps--flex"
                data-testid="namespace"
                style="width: 160px;"
              >
                <div
                  class="Layout--horizontal StyledProps--main DropDown--labelWrapper StyledProps--flex"
                >
                  <p
                    class="StyledProps--font Text--lineclamp StyledProps--main DropDown--label"
                    data-testid="dropdown-value"
                    style="--text-line-clamp: 1;"
                  >
                    common.namespace
                  </p>
                </div>
                <div
                  class="Layout--horizontal StyledProps--main DropDown--btnWrapper StyledProps--flex"
                >
                  <span
                    class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey400"
                    data-icon="main-chevron-down"
                  >
                    <svg
                      height="8"
                      viewBox="0 0 14 8"
                      width="8"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M13 0c-.3 0-.5.1-.7.3L7 5.6 1.7.3C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7l6 6c.2.2.4.3.7.3s.5-.1.7-.3l6-6c.2-.2.3-.4.3-.7 0-.6-.4-1-1-1z"
                        fill="#4d5969"
                      />
                    </svg>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <span
          style="flex-grow: 1;"
        />
        <div
          class="bp3-input-group ui-search-box ExpandingSearchInput--main   ExpandingSearchInput--alwaysExpanded ExpandingSearchInput--light"
          data-name=""
          style="width: 232px; max-width: 232px;"
        >
          <span
            class="bp3-icon StyledProps--main ExpandingSearchInput--icon StyledProps--color StyledProps--color-grey700"
            data-icon="thinner-search"
          >
            <svg
              fill="none"
              height="14"
              viewBox="0 0 16 16"
              width="14"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.125 13.25c1.617 0 3.32-.695 4.375-1.75L15 15l-3.5-3.5c1.021-1.005 1.75-2.894 1.75-4.375a6.125 6.125 0 10-6.125 6.125z"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </span>
          <input
            class="bp3-input  ExpandingSearchInput--filter"
            dir="auto"
            placeholder="discovery.searchService"
            style="padding-right: 38px;"
            type="search"
            value=""
          />
        </div>
        <div
          class="Layout--horizontal StyledProps--main StyledProps--flex"
        >
          <button
            class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main GridListToggle--gridButton StyledProps--intent-primary Button--with-current-color Button--without-shadow Button--withLeftIcon"
            data-testid="grid-view"
            data-tooltip-id="grid-view"
            type="button"
          >
            <span
              class="bp3-icon bp3-icon-grid-view StyledProps--main"
              icon="grid-view"
            >
              <svg
                data-icon="grid-view"
                height="16"
                viewBox="0 0 16 16"
                width="16"
              >
                <desc>
                  grid-view
                </desc>
                <path
                  d="M0 1v6h7V0H1C.45 0 0 .45 0 1zm0 14c0 .55.45 1 1 1h6V9H0v6zM15 0H9v7h7V1c0-.55-.45-1-1-1zM9 16h6c.55 0 1-.45 1-1V9H9v7z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </button>
          <button
            class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main GridListToggle--listUnselected GridListToggle--listButton Button--with-current-color Button--iconOnly Button--without-shadow Button--withLeftIcon"
            data-testid="list-view"
            data-tooltip-id="list-view"
            type="button"
          >
            <span
              class="bp3-icon bp3-icon-list StyledProps--main"
              icon="list"
            >
              <svg
                data-icon="list"
                height="16"
                viewBox="0 0 16 16"
                width="16"
              >
                <desc>
                  list
                </desc>
                <path
                  d="M1 3h14c.55 0 1-.45 1-1s-.45-1-1-1H1c-.55 0-1 .45-1 1s.45 1 1 1zm14 10H1c-.55 0-1 .45-1 1s.45 1 1 1h14c.55 0 1-.45 1-1s-.45-1-1-1zm0-4H1c-.55 0-1 .45-1 1s.45 1 1 1h14c.55 0 1-.45 1-1s-.45-1-1-1zm0-4H1c-.55 0-1 .45-1 1s.45 1 1 1h14c.55 0 1-.45 1-1s-.45-1-1-1z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="StyledProps--font StyledProps--main"
    >
      <div
        class="discoveredResourcesGraph"
      >
        Discovered Resources Graph
      </div>
    </div>
  </div>
</div>
`;

exports[`DiscoveredResources render component 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main"
  >
    <div
      class="Layout--horizontal StyledProps--main PageSubHeader--container StyledProps--flex"
      data-testid="page-subheader"
    >
      <div
        class="Layout--horizontal StyledProps--main StyledProps--flex"
        style="gap: 8px; width: 100%;"
      >
        <div
          class="bp3-popover-wrapper DropDown--main"
        >
          <div
            class="bp3-popover-target"
          >
            <div
              class=""
            >
              <div
                class="Layout--horizontal StyledProps--main DropDown--dropdownButton DropDown--withBorder StyledProps--flex"
                data-testid="namespace"
                style="width: 160px;"
              >
                <div
                  class="Layout--horizontal StyledProps--main DropDown--labelWrapper StyledProps--flex"
                >
                  <p
                    class="StyledProps--font Text--lineclamp StyledProps--main DropDown--label"
                    data-testid="dropdown-value"
                    style="--text-line-clamp: 1;"
                  >
                    common.namespace
                  </p>
                </div>
                <div
                  class="Layout--horizontal StyledProps--main DropDown--btnWrapper StyledProps--flex"
                >
                  <span
                    class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey400"
                    data-icon="main-chevron-down"
                  >
                    <svg
                      height="8"
                      viewBox="0 0 14 8"
                      width="8"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M13 0c-.3 0-.5.1-.7.3L7 5.6 1.7.3C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7l6 6c.2.2.4.3.7.3s.5-.1.7-.3l6-6c.2-.2.3-.4.3-.7 0-.6-.4-1-1-1z"
                        fill="#4d5969"
                      />
                    </svg>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <span
          style="flex-grow: 1;"
        />
        <div
          class="bp3-input-group ui-search-box ExpandingSearchInput--main   ExpandingSearchInput--alwaysExpanded ExpandingSearchInput--light"
          data-name=""
          style="width: 232px; max-width: 232px;"
        >
          <span
            class="bp3-icon StyledProps--main ExpandingSearchInput--icon StyledProps--color StyledProps--color-grey700"
            data-icon="thinner-search"
          >
            <svg
              fill="none"
              height="14"
              viewBox="0 0 16 16"
              width="14"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.125 13.25c1.617 0 3.32-.695 4.375-1.75L15 15l-3.5-3.5c1.021-1.005 1.75-2.894 1.75-4.375a6.125 6.125 0 10-6.125 6.125z"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </span>
          <input
            class="bp3-input  ExpandingSearchInput--filter"
            dir="auto"
            placeholder="discovery.searchService"
            style="padding-right: 38px;"
            type="search"
            value=""
          />
        </div>
        <div
          class="Layout--horizontal StyledProps--main StyledProps--flex"
        >
          <button
            class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main GridListToggle--gridButton StyledProps--intent-primary Button--with-current-color Button--without-shadow Button--withLeftIcon"
            data-testid="grid-view"
            data-tooltip-id="grid-view"
            type="button"
          >
            <span
              class="bp3-icon bp3-icon-grid-view StyledProps--main"
              icon="grid-view"
            >
              <svg
                data-icon="grid-view"
                height="16"
                viewBox="0 0 16 16"
                width="16"
              >
                <desc>
                  grid-view
                </desc>
                <path
                  d="M0 1v6h7V0H1C.45 0 0 .45 0 1zm0 14c0 .55.45 1 1 1h6V9H0v6zM15 0H9v7h7V1c0-.55-.45-1-1-1zM9 16h6c.55 0 1-.45 1-1V9H9v7z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </button>
          <button
            class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main GridListToggle--listUnselected GridListToggle--listButton Button--with-current-color Button--iconOnly Button--without-shadow Button--withLeftIcon"
            data-testid="list-view"
            data-tooltip-id="list-view"
            type="button"
          >
            <span
              class="bp3-icon bp3-icon-list StyledProps--main"
              icon="list"
            >
              <svg
                data-icon="list"
                height="16"
                viewBox="0 0 16 16"
                width="16"
              >
                <desc>
                  list
                </desc>
                <path
                  d="M1 3h14c.55 0 1-.45 1-1s-.45-1-1-1H1c-.55 0-1 .45-1 1s.45 1 1 1zm14 10H1c-.55 0-1 .45-1 1s.45 1 1 1h14c.55 0 1-.45 1-1s-.45-1-1-1zm0-4H1c-.55 0-1 .45-1 1s.45 1 1 1h14c.55 0 1-.45 1-1s-.45-1-1-1zm0-4H1c-.55 0-1 .45-1 1s.45 1 1 1h14c.55 0 1-.45 1-1s-.45-1-1-1z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="StyledProps--font StyledProps--main"
    >
      <div
        class="discoveredResourcesGraph"
      >
        Discovered Resources Graph
      </div>
    </div>
  </div>
</div>
`;

exports[`DiscoveredResources subheader components should be present 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main"
  >
    <div
      class="Layout--horizontal StyledProps--main PageSubHeader--container StyledProps--flex"
      data-testid="page-subheader"
    >
      <div
        class="Layout--horizontal StyledProps--main StyledProps--flex"
        style="gap: 8px; width: 100%;"
      >
        <div
          class="bp3-popover-wrapper DropDown--main"
        >
          <div
            class="bp3-popover-target"
          >
            <div
              class=""
            >
              <div
                class="Layout--horizontal StyledProps--main DropDown--dropdownButton DropDown--withBorder StyledProps--flex"
                data-testid="namespace"
                style="width: 160px;"
              >
                <div
                  class="Layout--horizontal StyledProps--main DropDown--labelWrapper StyledProps--flex"
                >
                  <p
                    class="StyledProps--font Text--lineclamp StyledProps--main DropDown--label"
                    data-testid="dropdown-value"
                    style="--text-line-clamp: 1;"
                  >
                    common.namespace
                  </p>
                </div>
                <div
                  class="Layout--horizontal StyledProps--main DropDown--btnWrapper StyledProps--flex"
                >
                  <span
                    class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey400"
                    data-icon="main-chevron-down"
                  >
                    <svg
                      height="8"
                      viewBox="0 0 14 8"
                      width="8"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M13 0c-.3 0-.5.1-.7.3L7 5.6 1.7.3C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7l6 6c.2.2.4.3.7.3s.5-.1.7-.3l6-6c.2-.2.3-.4.3-.7 0-.6-.4-1-1-1z"
                        fill="#4d5969"
                      />
                    </svg>
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="bp3-overlay bp3-overlay-inline"
          >
            <div
              class="bp3-transition-container bp3-popover-exit bp3-popover-exit-active"
              style="position: absolute; will-change: transform; top: 0px; left: 0px; transform: translate3d(NaNpx, NaNpx, 0);"
            >
              <div
                class="bp3-popover bp3-minimal bp3-select-popover DropDown--popover"
                style="transform-origin: center top;"
              >
                <div
                  class="bp3-popover-content"
                >
                  <div>
                    <div
                      class="bp3-input-group"
                    >
                      <span
                        class="bp3-input-left-container"
                      >
                        <span
                          class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey500"
                          data-icon="thinner-search"
                        >
                          <svg
                            fill="none"
                            height="12"
                            viewBox="0 0 16 16"
                            width="12"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M7.125 13.25c1.617 0 3.32-.695 4.375-1.75L15 15l-3.5-3.5c1.021-1.005 1.75-2.894 1.75-4.375a6.125 6.125 0 10-6.125 6.125z"
                              stroke="currentColor"
                              stroke-linecap="round"
                              stroke-linejoin="round"
                            />
                          </svg>
                        </span>
                      </span>
                      <input
                        class="bp3-input"
                        placeholder="Search"
                        style="padding-left: 0px;"
                        type="text"
                        value=""
                      />
                    </div>
                    <ul
                      class="bp3-menu"
                    >
                      <li
                        class="DropDown--menuItem DropDown--active"
                      >
                        All
                      </li>
                      <li
                        class="DropDown--menuItem"
                      >
                        boutique
                      </li>
                    </ul>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <span
          style="flex-grow: 1;"
        />
        <div
          class="bp3-input-group ui-search-box ExpandingSearchInput--main   ExpandingSearchInput--alwaysExpanded ExpandingSearchInput--light"
          data-name=""
          style="width: 232px; max-width: 232px;"
        >
          <span
            class="bp3-icon StyledProps--main ExpandingSearchInput--icon StyledProps--color StyledProps--color-grey700"
            data-icon="thinner-search"
          >
            <svg
              fill="none"
              height="14"
              viewBox="0 0 16 16"
              width="14"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.125 13.25c1.617 0 3.32-.695 4.375-1.75L15 15l-3.5-3.5c1.021-1.005 1.75-2.894 1.75-4.375a6.125 6.125 0 10-6.125 6.125z"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </span>
          <input
            class="bp3-input  ExpandingSearchInput--filter"
            dir="auto"
            placeholder="discovery.searchService"
            style="padding-right: 38px;"
            type="search"
            value="test abc"
          />
          <span
            class="ExpandingSearchInput--btnWrapper"
          >
            <button
              class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main Button--with-current-color Button--iconOnly Button--without-shadow Button--withLeftIcon"
              type="button"
            >
              <span
                class="bp3-icon StyledProps--main"
                data-icon="main-close"
              >
                <svg
                  height="8"
                  viewBox="0 0 14 14"
                  width="8"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8.6 7l5-5c.3-.2.4-.5.4-.8 0-.7-.5-1.2-1.2-1.2-.3 0-.6.1-.8.3l-5 5-5-5C1.8.1 1.5 0 1.2 0 .5 0 0 .5 0 1.2c0 .3.1.6.3.8l5 5-5 5c-.2.2-.3.5-.3.8 0 .7.5 1.2 1.2 1.2.3 0 .6-.1.8-.3l5-5 5 5c.2.2.5.4.8.4.6 0 1.2-.5 1.2-1.2 0-.3-.1-.6-.3-.8z"
                    fill="#4d5969"
                  />
                </svg>
              </span>
            </button>
          </span>
        </div>
        <div
          class="Layout--horizontal StyledProps--main StyledProps--flex"
        >
          <button
            class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main GridListToggle--gridButton StyledProps--intent-primary Button--with-current-color Button--without-shadow Button--withLeftIcon"
            data-testid="grid-view"
            data-tooltip-id="grid-view"
            type="button"
          >
            <span
              class="bp3-icon bp3-icon-grid-view StyledProps--main"
              icon="grid-view"
            >
              <svg
                data-icon="grid-view"
                height="16"
                viewBox="0 0 16 16"
                width="16"
              >
                <desc>
                  grid-view
                </desc>
                <path
                  d="M0 1v6h7V0H1C.45 0 0 .45 0 1zm0 14c0 .55.45 1 1 1h6V9H0v6zM15 0H9v7h7V1c0-.55-.45-1-1-1zM9 16h6c.55 0 1-.45 1-1V9H9v7z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </button>
          <button
            class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main GridListToggle--listUnselected GridListToggle--listButton Button--with-current-color Button--iconOnly Button--without-shadow Button--withLeftIcon"
            data-testid="list-view"
            data-tooltip-id="list-view"
            type="button"
          >
            <span
              class="bp3-icon bp3-icon-list StyledProps--main"
              icon="list"
            >
              <svg
                data-icon="list"
                height="16"
                viewBox="0 0 16 16"
                width="16"
              >
                <desc>
                  list
                </desc>
                <path
                  d="M1 3h14c.55 0 1-.45 1-1s-.45-1-1-1H1c-.55 0-1 .45-1 1s.45 1 1 1zm14 10H1c-.55 0-1 .45-1 1s.45 1 1 1h14c.55 0 1-.45 1-1s-.45-1-1-1zm0-4H1c-.55 0-1 .45-1 1s.45 1 1 1h14c.55 0 1-.45 1-1s-.45-1-1-1zm0-4H1c-.55 0-1 .45-1 1s.45 1 1 1h14c.55 0 1-.45 1-1s-.45-1-1-1z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="StyledProps--font StyledProps--main"
    >
      <div
        class="discoveredResourcesGraph"
      >
        Discovered Resources Graph
      </div>
    </div>
  </div>
</div>
`;

exports[`DiscoveredResources when namespace data is undefined 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main"
  >
    <div
      class="Layout--horizontal StyledProps--main PageSubHeader--container StyledProps--flex"
      data-testid="page-subheader"
    >
      <div
        class="Layout--horizontal StyledProps--main StyledProps--flex"
        style="gap: 8px; width: 100%;"
      >
        <div
          class="bp3-popover-wrapper DropDown--main"
        >
          <div
            class="bp3-popover-target"
          >
            <div
              class=""
            >
              <div
                class="Layout--horizontal StyledProps--main DropDown--dropdownButton DropDown--withBorder StyledProps--flex"
                data-testid="namespace"
                style="width: 160px;"
              >
                <div
                  class="Layout--horizontal StyledProps--main DropDown--labelWrapper StyledProps--flex"
                >
                  <p
                    class="StyledProps--font Text--lineclamp StyledProps--main DropDown--label"
                    data-testid="dropdown-value"
                    style="--text-line-clamp: 1;"
                  >
                    common.namespace
                  </p>
                </div>
                <div
                  class="Layout--horizontal StyledProps--main DropDown--btnWrapper StyledProps--flex"
                >
                  <span
                    class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey400"
                    data-icon="main-chevron-down"
                  >
                    <svg
                      height="8"
                      viewBox="0 0 14 8"
                      width="8"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M13 0c-.3 0-.5.1-.7.3L7 5.6 1.7.3C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7l6 6c.2.2.4.3.7.3s.5-.1.7-.3l6-6c.2-.2.3-.4.3-.7 0-.6-.4-1-1-1z"
                        fill="#4d5969"
                      />
                    </svg>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <span
          style="flex-grow: 1;"
        />
        <div
          class="bp3-input-group ui-search-box ExpandingSearchInput--main   ExpandingSearchInput--alwaysExpanded ExpandingSearchInput--light"
          data-name=""
          style="width: 232px; max-width: 232px;"
        >
          <span
            class="bp3-icon StyledProps--main ExpandingSearchInput--icon StyledProps--color StyledProps--color-grey700"
            data-icon="thinner-search"
          >
            <svg
              fill="none"
              height="14"
              viewBox="0 0 16 16"
              width="14"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.125 13.25c1.617 0 3.32-.695 4.375-1.75L15 15l-3.5-3.5c1.021-1.005 1.75-2.894 1.75-4.375a6.125 6.125 0 10-6.125 6.125z"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </span>
          <input
            class="bp3-input  ExpandingSearchInput--filter"
            dir="auto"
            placeholder="discovery.searchService"
            style="padding-right: 38px;"
            type="search"
            value=""
          />
        </div>
        <div
          class="Layout--horizontal StyledProps--main StyledProps--flex"
        >
          <button
            class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main GridListToggle--gridButton StyledProps--intent-primary Button--with-current-color Button--without-shadow Button--withLeftIcon"
            data-testid="grid-view"
            data-tooltip-id="grid-view"
            type="button"
          >
            <span
              class="bp3-icon bp3-icon-grid-view StyledProps--main"
              icon="grid-view"
            >
              <svg
                data-icon="grid-view"
                height="16"
                viewBox="0 0 16 16"
                width="16"
              >
                <desc>
                  grid-view
                </desc>
                <path
                  d="M0 1v6h7V0H1C.45 0 0 .45 0 1zm0 14c0 .55.45 1 1 1h6V9H0v6zM15 0H9v7h7V1c0-.55-.45-1-1-1zM9 16h6c.55 0 1-.45 1-1V9H9v7z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </button>
          <button
            class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main GridListToggle--listUnselected GridListToggle--listButton Button--with-current-color Button--iconOnly Button--without-shadow Button--withLeftIcon"
            data-testid="list-view"
            data-tooltip-id="list-view"
            type="button"
          >
            <span
              class="bp3-icon bp3-icon-list StyledProps--main"
              icon="list"
            >
              <svg
                data-icon="list"
                height="16"
                viewBox="0 0 16 16"
                width="16"
              >
                <desc>
                  list
                </desc>
                <path
                  d="M1 3h14c.55 0 1-.45 1-1s-.45-1-1-1H1c-.55 0-1 .45-1 1s.45 1 1 1zm14 10H1c-.55 0-1 .45-1 1s.45 1 1 1h14c.55 0 1-.45 1-1s-.45-1-1-1zm0-4H1c-.55 0-1 .45-1 1s.45 1 1 1h14c.55 0 1-.45 1-1s-.45-1-1-1zm0-4H1c-.55 0-1 .45-1 1s.45 1 1 1h14c.55 0 1-.45 1-1s-.45-1-1-1z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="StyledProps--font StyledProps--main"
    >
      <div
        class="discoveredResourcesGraph"
      >
        Discovered Resources Graph
      </div>
    </div>
  </div>
</div>
`;
